test/saithrift/tests/sail3.py

```diff
@@ -1214,3 +1214,116 @@ def runTest(self):
+        attr_value1 = sai_thrift_attribute_value_t(u16=vlan_id)
+        attr1 = sai_thrift_attribute_t(id=SAI_PORT_ATTR_PORT_VLAN_ID, value=attr_value1)
+        self.client.sai_thrift_set_port_attribute(port1, attr1)
+        self.client.sai_thrift_set_port_attribute(port2, attr1)
+        
+        vr_id = sai_thrift_create_virtual_router(self.client, v4_enabled, v6_enabled)
+        rif_vlan_id = sai_thrift_create_router_interface(self.client, vr_id, 0, 0, vlan_id, v4_enabled, v6_enabled, mac1)
+        rif_port_id = sai_thrift_create_router_interface(self.client, vr_id, 1, port3, 0, v4_enabled, v6_enabled, mac2)
+
+        attr_value2 = sai_thrift_attribute_value_t(s32=SAI_PORT_LEARN_MODE_HW)
+        attr2 = sai_thrift_attribute_t(id=SAI_PORT_ATTR_FDB_LEARNING, value=attr_value2)
+        self.client.sai_thrift_set_port_attribute(port1, attr2)
+        self.client.sai_thrift_set_port_attribute(port2, attr2)
+        
+        sai_thrift_create_neighbor(self.client, addr_family, rif_vlan_id, ip_port1, mac_port1)
+        sai_thrift_create_route(self.client, vr_id, addr_family, ip_addr1_subnet, ip_mask1, rif_vlan_id)
+            
+        
+        local_pkt = simple_tcp_packet(eth_dst='00:22:22:22:22:22',
+                                eth_src=mac_port1,
+                                dl_vlan_enable=True,
+                                vlan_vid=vlan_id,
+                                ip_dst='10.0.0.1',
+                                ip_src=ip_port1,
+                                ip_id=102,
+                                ip_ttl=64)
+        
+
+        try:
+            #sending unkown UC for learning the ports mac and expecting flooding only on the vlan
+            send_packet(self, 0, str(local_pkt))
+            verify_packets(self, local_pkt, [1])
+            verify_no_other_packets(self)
+            #sending L3 packet from port 3 through router to port 1 that update the fdb with is MAC
+            L3_pkt = simple_tcp_packet(pktlen=100,
+                                eth_dst=router_mac,
+                                eth_src=mac_port3,
+                                ip_src='11.11.11.1',
+                                ip_dst=ip_port1,
+                                ip_id=105,
+                                ip_ttl=64)
+            exp_pkt = simple_tcp_packet(pktlen=104,#additional 4bytes because of the vlan
+                                eth_dst=mac_port1,
+                                eth_src=router_mac,
+                                ip_dst=ip_port1,
+                                ip_src='11.11.11.1',
+                                dl_vlan_enable=True,
+                                vlan_vid=vlan_id,
+                                ip_id=105,
+                                ip_ttl=63)
+            send_packet(self, 2, str(L3_pkt))
+            verify_packets(self, exp_pkt, [0])
+            verify_no_other_packets(self)
```

Comment:
No need for verify_no_other_packets

verify_packets :     """
    Check that a packet is received on each of the specified port numbers for a
    given device (default device number is 0).
    Also verifies that the packet is not received on any other ports for this
    device, and that no other packets are received on the device (unless --relax
    is in effect).